src/connector/src/parser/common.rs

```diff
@@ -24,42 +26,67 @@ use risingwave_expr::vector_op::cast::{
-            ensure_int!(v, i32)
-                .try_into()
-                .map_err(|e| anyhow!("expect i32: {}", e))?,
-        ),
-        DataType::Int64 => ensure_int!(v, i64).into(),
-        DataType::Float32 => ScalarImpl::Float32((simd_json_ensure_float!(v, f32) as f32).into()),
-        DataType::Float64 => ScalarImpl::Float64((simd_json_ensure_float!(v, f64)).into()),
+        DataType::Boolean => match v {
+            BorrowedValue::Static(StaticNode::Bool(b)) => (*b).into(),
+            // debezium converts bool to int, false -> 0, true -> 1, for mysql and postgres
+            BorrowedValue::Static(v) => match v.as_i64() {
+                Some(0i64) => ScalarImpl::Bool(false),
+                Some(1i64) => ScalarImpl::Bool(true),
+                _ => anyhow::bail!("expect bool, but found {v}"),
+            },
+            _ => anyhow::bail!("expect bool, but found {v}"),
+        },
+        DataType::Int16 => ensure_i16!(v, i16).into(),
+        DataType::Int32 => ensure_i32!(v, i32).into(),
+        DataType::Int64 => ensure_i64!(v, i64).into(),
+        // when f32 overflows, the value is converted to `inf` which is inappropriate
+        DataType::Float32 => {
+            let scalar_val = ScalarImpl::Float32((simd_json_ensure_float!(v, f32) as f32).into());
+            if let ScalarImpl::Float32(f) = scalar_val {
+                if f.is_infinite() {
+                    anyhow::bail!("{v} is out of range for type f32");
+                }
+            }
+            scalar_val
+        }
+        DataType::Float64 => simd_json_ensure_float!(v, f64).into(),
         // FIXME: decimal should have more precision than f64
         DataType::Decimal => Decimal::from_f64(simd_json_ensure_float!(v, Decimal))
             .ok_or_else(|| anyhow!("expect decimal"))?
             .into(),
         DataType::Varchar => ensure_str!(v, "varchar").to_string().into(),
         DataType::Bytea => ensure_str!(v, "bytea").to_string().into(),
-        DataType::Date => str_to_date(ensure_str!(v, "date"))?.into(),
-        DataType::Time => str_to_time(ensure_str!(v, "time"))?.into(),
+        // debezium converts date to i32 for mysql and postgres
+        DataType::Date => match v {
+            BorrowedValue::String(s) => str_to_date(s)?.into(),
+            BorrowedValue::Static(_) => {
+                NaiveDateWrapper::with_days_since_unix_epoch(ensure_i32!(v, i32))?.into()
+            }
+            _ => anyhow::bail!("expect date, but found {v}"),
+        },
+        // debezium converts time to i64 for mysql and postgres
+        DataType::Time => match v {
+            BorrowedValue::String(_) => str_to_time(ensure_str!(v, "time"))?.into(),
+            BorrowedValue::Static(_) => NaiveTimeWrapper::with_milli(
+                ensure_i64!(v, i64)
+                    .try_into()
+                    .map_err(|_| anyhow!("cannot cast i64 to time, value out of range"))?,
+            )?
+            .into(),
```

Comment:
Is it always `milli`?

https://debezium.io/documentation/reference/stable/connectors/mysql.html#mysql-temporal-types
https://debezium.io/documentation/reference/stable/connectors/postgresql.html#postgresql-basic-types

According to the docs, `time.precision.mode=connect` uses `milli` but `time.precision.mode=adaptive_time_microseconds` uses `micro`.